tests/test_communication_utils.cpp

```diff
@@ -0,0 +1,196 @@
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "config.h"
+
+#define BOOST_TEST_MODULE CommunicationUtilities
+#include <boost/test/unit_test.hpp>
+#include <boost/test/unit_test.hpp>
+#include <dune/common/parallel/mpihelper.hh>
+#include <opm/common/utility/CommunicationUtils.hpp>
+#include <vector>
+#include <random>
+
+class MPIError {
+public:
+  /** @brief Constructor. */
+  MPIError(std::string s, int e) : errorstring(s), errorcode(e){}
+  /** @brief The error string. */
+  std::string errorstring;
+  /** @brief The mpi error code. */
+  int errorcode;
+};
+
+#ifdef HAVE_MPI
+void MPI_err_handler(MPI_Comm *, int *err_code, ...){
+  char *err_string=new char[MPI_MAX_ERROR_STRING];
+  int err_length;
+  MPI_Error_string(*err_code, err_string, &err_length);
+  std::string s(err_string, err_length);
+  std::cerr << "An MPI Error ocurred:"<<std::endl<<s<<std::endl;
+  delete[] err_string;
+  throw MPIError(s, *err_code);
+}
+#endif
+
+struct MPIFixture
+{
+    MPIFixture()
+    {
+#if HAVE_MPI
+    int m_argc = boost::unit_test::framework::master_test_suite().argc;
+    char** m_argv = boost::unit_test::framework::master_test_suite().argv;
+    helper = &Dune::MPIHelper::instance(m_argc, m_argv);
+#ifdef MPI_2
+    MPI_Comm_create_errhandler(MPI_err_handler, &handler);
+   MPI_Comm_set_errhandler(MPI_COMM_WORLD, handler);
+#else
+        MPI_Errhandler_create(MPI_err_handler, &handler);
+        MPI_Errhandler_set(MPI_COMM_WORLD, handler);
+#endif
+#endif
+    }
+    ~MPIFixture()
+    {
+#if HAVE_MPI
+        MPI_Finalize();
+#endif
+    }
+    Dune::MPIHelper* helper;
+#if HAVE_MPI
+    MPI_Errhandler handler;
+#endif
+};
+
+BOOST_GLOBAL_FIXTURE(MPIFixture);
+
+template<class T, class C>
+std::tuple<std::vector<T>, std::vector<T>, std::vector<int>, std::vector<int>>
+createParallelData(const C& comm, T)
+{
+    std::vector<int> sizes(comm.size());
+    std::vector<int> displ(comm.size()+1);
+    std::vector<T> allVals;
+    
+    if (comm.rank() == 0)
+    {
+        // Initialize with random numbers.
+        std::random_device rndDevice;
+        std::mt19937 mersenneEngine {rndDevice()};  // Generates random integers
+        std::uniform_int_distribution<int> dist {1, 10};
+        
+        auto gen = [&dist, &mersenneEngine](){
+                       return dist(mersenneEngine);
+                   };
+        auto begin = std::begin(sizes);
+        auto end = std::end(sizes);
+        std::generate(begin, end, gen);
+        std::partial_sum(begin, end, std::begin(displ)+1);
```

Comment:
This needs
```C++
#include <numeric>
```

Reply:
Shouldn't the include of that header in opm/grid/common/CommunicationUtils.hpp be enough?

Reply:
Added nevertheless...